src/availability.rs

```diff
@@ -325,6 +325,44 @@ where
                     .unwrap())
             }
             .boxed()
+        })?
+        .get("get_block_summary", move |req, state| {
+            async move {
+                let id: usize = req.integer_param("height")?;
+
+                state
+                    .get_block(BlockId::Number(id))
+                    .await
+                    .with_timeout(timeout)
+                    .await
+                    .context(FetchBlockSnafu {
+                        resource: id.to_string(),
+                    })
+                    .map(Into::<BlockSummaryQueryData<Types>>::into)
+            }
+            .boxed()
+        })?
+        .get("get_block_summary_range", move |req, state| {
+            async move {
+                let from: usize = req.integer_param("from")?;
+                let until: usize = req.integer_param("until")?;
+
+                let result: Vec<BlockSummaryQueryData<Types>> = state
+                    .get_block_range(from..until)
+                    .await
+                    .enumerate()
+                    .map(|(index, fetch)| {
+                        fetch.context(FetchBlockSnafu {
```

Comment:
I think to be consistent with the rest of the API, we should use `fetch.with_timeout` here (ie fail if we don't get the block after a short time, rather than failing immediately). This will require an `await` in the closure here, so you can use `.then` instead of `.map` to make the closure `async`

Reply:
I initially tried chaining a `.with_timeout` off of the `get_block_range` off of this call, but it doesn't seem to exist on the `get_block_range` call. My guess is because it returns a `Stream` rather than a `Future`. I **could** add the `timeout` in the `map` call, where the `context` call is made if that makes more sense.

Additionally because this is a `Stream` of values, I don't think I can actually call `then` on the result.

Reply:
Ah I see.  That's a really weird `API` definition